frontend/src/pages/BookGalleryPage/bookGalleryPage/BookGalleryPage.tsx

```diff
@@ -16,13 +17,59 @@ export default function BookGalleryPage({filteredBooks, setSearchInput}: BookGal
     const [statusFilter, setStatusFilter] = useState<string>("ALL")
 
 
+
+    const [showFilter, setShowFilter] = useState<boolean>(false);
+    const [selectedGenre, setSelectedGenre] = useState<string>("Select");
+    const [showFilterTag, setShowFilterTag] = useState<boolean>(false);
+
+    const handleClick = () => {
+        setShowFilter(!showFilter);
+    }
+
+    const handleApplyFilter = (genre: string) => {
+        setShowFilter(false);
+        console.log(genre);
+        setShowFilterTag(genre !== "Select");
+    }
+
+    const handleRemoveFilter = () => {
+        setShowFilterTag(false);
+        setSelectedGenre('Select');
+    }
+
     return (
         <div id={"galleryPage"}>
             <SearchBar setSearchInput={setSearchInput}/>
+            <div className={"filter-sector"}>
+                <button
+                    onClick={handleClick}
+                    className={"show-filter-button"}
+                >
+                    {!showFilter ? "Advanced Search" : "x"}
+                </button>
+                {showFilter &&
+                    <FilterPage
+                        selectedGenre={selectedGenre}
+                        setSelectedGenre={setSelectedGenre}
+                        handleApplyFilter={handleApplyFilter}
+                        setShowKeywordTag={setShowFilterTag}
+                    />
+                }
+                {(showFilterTag && !showFilter) &&
+                    <div className={"filter-tag-area"}>
+                        <p style={{fontStyle: "italic", fontWeight: "bold"}}>Filtered Genre: </p>
+                        <div className={"filter-tag"}>
+                            <p className={"filter-tag-genre"}>{formatEnum(selectedGenre)}</p>
+                            <button className={"filter-tag-close"} onClick={handleRemoveFilter}>x</button>
```

Comment:
would make the filter name part of the button, or make it clearer, that only the x is the button
```suggestion
                            <button className={"filter-tag-close"} onClick={handleRemoveFilter}>{formatEnum(selectedGenre)} x</button>
```

Reply:
erledgit